### PR TITLE
Android build fix

### PR DESCRIPTION
**Pull request by linusmartensson**
Since OS is transformed from android to linux, we need to ensure that conditions
specifying OS==linux are prevented from executing by additionally specifying
'and android_build=0'. 

This was done on most conditions in bnoordhuis patch, but I added a couple missed ones that broke the android build.

### CODE REVIEW

File: common.gypi [hunk 1]
```diff
@@ -143,7 +143,7 @@
             'cflags': [ '-m32' ],
             'ldflags': [ '-m32' ],
           }],
-          [ 'OS=="linux"', {
+          [ 'OS=="linux" and android_build == 0', {
             'cflags': [ '-ansi' ],
```
Comment (bnoordhuis) [inline common.gypi:146]: What does -ansi break (and why)?

File: uv.gyp [hunk 2]
```diff
@@ -51,7 +51,7 @@
           ['OS == "mac"', {
             'defines': [ '_DARWIN_USE_64_BIT_INODE=1' ],
           }],
-          ['OS == "linux"', {
+          ['OS == "linux" and android_build == 0', {
             'defines': [ '_POSIX_C_SOURCE=200112' ],
```
Comment (bnoordhuis) [inline uv.gyp:54]: Ditto. Should be harmless, right?

**Comment by bnoordhuis** [comment]
I'm okay with the patch but please update the commit log. Wasn't there a Makefile fix-up as well? -lrt?

**Comment by linusmartensson** [comment]
This is, I hope, a bit clearer.

**Comment by bnoordhuis** [comment]
Thanks, landed in ff3ee84.

**Pull request closed without merging**